### PR TITLE
feat: Add cdf method to AsymptoticTestStatDistribution

### PR DESCRIPTION
**Pull request by lukasheinrich**
# Description

This adds a cdf method to the asymptotic distribution. This will be useful for two-sided tests. c.f. discussion in PR #966.

ReadTheDocs build: https://pyhf.readthedocs.io/en/asymptotic_cdf

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR

```
* Add cumulative distribution function method to AsymptoticTestStatDistribution
* Add tests for test stat distribution calculators
```

### CODE REVIEW

File: src/pyhf/infer/calculators.py [hunk 1]
```diff
@@ -59,6 +59,20 @@ def __init__(self, shift):
         self.shift = shift
         self.sqrtqmuA_v = None
 
+    def cdf(self, value):
+        """
+        Compute the value of the cumulative distribution funcation for a given value of the test statistic.
```
Comment (kratsg) [inline src/pyhf/infer/calculators.py:64]:
```suggestion
        Compute the value of the cumulative distribution function for a given value of the test statistic.
```

File: src/pyhf/infer/calculators.py [hunk 2]
```diff
@@ -59,6 +59,20 @@ def __init__(self, shift):
...
+
+        Args:
+            value (`float`): The test statistic value.
+
+        Returns:
+            Float: The integrated probability to observe a test statistic less than or equal to the observed ``value``.
```
Comment (kratsg) [inline src/pyhf/infer/calculators.py:70]:
```suggestion
            probability (`float`): The integrated probability to observe a test statistic less than or equal to the observed ``value``.
```

File: tests/test_calculator.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import pyhf
+import pyhf.infer.calculators
+
+
+def test_calc_dist():
+    asymptotic_dist = pyhf.infer.calculators.AsymptoticTestStatDistribution(0.0)
+    assert asymptotic_dist.pvalue(1) == asymptotic_dist.cdf(-1)
```
Comment (kratsg) [inline tests/test_calculator.py:7]: why is this true? If AsymptoticTestStatDistribution has `0.0 = -u^/sigma` with variance 1.0, then...>?

File: src/pyhf/infer/calculators.py (hunk 1 above)
Comment (matthewfeickert) [reply src/pyhf/infer/calculators.py]: I think that might have been my bad. Sorry.

File: src/pyhf/infer/calculators.py [hunk 4]
```diff
@@ -59,6 +59,20 @@ def __init__(self, shift):
...
 
+    def cdf(self, value):
+        """
+        Compute the value of the cumulative distribution function for a given value of the test statistic.
+
+        Args:
+            value (`float`): The test statistic value.
+
+        Returns:
+            probability (`float`): The integrated probability to observe a test statistic less than or equal to the observed ``value``.
```
Comment (matthewfeickert) [inline src/pyhf/infer/calculators.py:70]:
```suggestion
            Float: The integrated probability to observe a test statistic less than or equal to the observed ``value``.
```

Shouldn't this be the return type? This is how it is for all the rest of our docs. e.g. https://scikit-hep.org/pyhf/_generated/pyhf.tensor.pytorch_backend.pytorch_backend.html#pyhf.tensor.pytorch_backend.pytorch_backend.astensor

**Pull request merged**